manifests/charts/istiocoredns/values.yaml

```diff
@@ -99,6 +99,16 @@ global:
   imagePullSecrets: []
   # - private-registry-key
 
+  # Custom DNS config for the pod to resolve names of services in other
+  # clusters. Use this to add additional search domains, and other settings.
+  # see
+  # https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#dns-config
+  # This does not apply to gateway pods as they typically need a different
+  # set of DNS settings than the normal application pods (e.g., in
+  # multicluster scenarios).
+  podDNSSearchNamespaces:
+  - global
```

Comment:
do you ever expect this list to expand? There is a general problem with this approach in that K8s has a limit of 6 search locations in /etc/resolv.conf. This makes expanding this list difficult, and in general even using this list difficult. The reason is that many cloud providers already use 4 to 5 search spaces, so adding 1, while not always bad, may not always work. Adding 2 or more can crater the entire deployment.

Reply:
yes I expect the list to expand. what do you mean the limitation of resolv.conf. we can patch the coredns/kubedns configmap in kube-system to proxy to istiocoredns

Reply:
https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L416-L421 will cause every sidecar's resolv.conf to be injected with DNSConfig's list. I know - because I wrote the code...

https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config

K8s "allows for at most 6 search domains". Unfortunately, I would later find out this is baked into glibc itself.

Hence, why there is no variable list, but only `.global`. There was, at one time, a variable list, but it broke most of the time on most of the clouds.

Cheers,
-steve

Reply:
@sdake Okay so the DNSSearchNamespaces is injected to the sidecar. but how it will affect the multicluster communication? since for my understanding, the app pod make a call directly to coredns/kubedns in kube-system to resolve the domain.